Vc/avx/const.h

```diff
@@ -119,8 +119,7 @@ namespace AVX
     {
 #ifdef Vc_IMPL_AVX2
 #if defined Vc_ICC || defined Vc_MSVC
-        __m256i allone;
-        allone = _mm256_cmpeq_epi8(allone, allone);
+        __m256i allone = _mm256_cmpeq_epi8(__m256i(), __m256i());
```

Comment:
@amadio, how much worse is the codegen now? https://godbolt.org/z/E98Wrr7E7 A lot more instructions are produced now.

Reply:
Argh, compilers suck, let's go with this then: https://godbolt.org/z/s7qazEhd9
```cpp
return _mm256_xor_epi64(_mm256_setzero_si256(), _mm256_setzero_si256());
```

Reply:
Compilers suck, let's go with the `_mm256_set1_epi64x` call, but use `~0` instead of `-1`?

Reply:
@bernhardmgruber, it looks like we ran into yet another case of https://developercommunity.visualstudio.com/t/Missed-optimization:-coalescing-contiguo/1471162 . But this one shows MSVC deoptimizes the whole function just by running into a register initialization? (Also, a small typo on your test case: the architecture flag in MSVC is `/arch`, you used the Clang/GCC version.)

@amadio complete test case here: https://godbolt.org/z/cbaGhPW4K. Any of the options (~0, -1, calling numeric_limits) will suffice.